ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java

```diff
@@ -418,10 +418,8 @@ private static Optional<Integer> ensureKeySchemasMatch(
     }
 
     if (!latest.isPresent()) {
-      // Note: This will result in auto-registering a key schema if the schema is not found from SR
-      // This is a known issue and, we will tackle this in a separate issue once we have determined
-      // whether we want to continue auto-registering during INSERT
-      return Optional.empty();
+      throw new KsqlException("Failed to fetch schema. Please check if schema exists in "
```

Comment:
Can you include the subject name the user can use to check? i.e. would the user use `topic-key` or `topic-value`?

Could you also specify `Failed to fetch the key schema.`?

Reply:
Done